components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java

```diff
@@ -543,6 +568,35 @@ private void storeSessionData(AuthenticationContext context, String sessionConte
                         "user store domain: " + userStoreDomain + " in tenant domain: " + tenantDomain , e);
             }
         }
+
+        try {
+            if (appId != -1 && !UserSessionStore.getInstance().isExistingAppSession(sessionContextKey, subject,
```

Comment:
Better to move this -1 to a constant with a descriptive name.

Reply:
Updated